### PR TITLE
A21 update: Add note on fallback to default service config if no service config is found

### PR DESCRIPTION
**Pull request by yashykt**
Add note on fallback to default service config if no service config is found

### CODE REVIEW

File: A21-service-config-error-handling.md [hunk 1]
```diff
@@ -44,8 +44,11 @@ if the chosen service config is found invalid while the other service configs
 were valid, the gRPC config would still be considered invalid as a whole.
 2. The client validates the choice based on the criteria outlined in the
 section below.
-3. If the service config is valid, the client uses the configuration provided.
-4. If the service config is invalid, the clients should reject the received
+3. If no service config is found, the client falls back on the default service
+config. If there is no default service configured either, the client selects no
```
Comment (markdroth) [inline A21-service-config-error-handling.md:48]: I think if there is no default service config set for the channel, then it will use the an empty config.

**Comment by markdroth** [comment]
@ejona86 and @dfawley should also review, just to make sure I haven't missed anything.

File: A21-service-config-error-handling.md [hunk 2]
```diff
@@ -44,8 +44,11 @@ if the chosen service config is found invalid while the other service configs
 were valid, the gRPC config would still be considered invalid as a whole.
 2. The client validates the choice based on the criteria outlined in the
 section below.
-3. If the service config is valid, the client uses the configuration provided.
-4. If the service config is invalid, the clients should reject the received
+3. If no service config is found, the client falls back on the default service
```
Comment (dfawley) [inline A21-service-config-error-handling.md:47]: Should we clarify this a little and say "the default service config _set via the client/channel API_" (e.g.)?  I don't think we've ever documented what the "default service config" is or where it comes from.  (It is already referenced elsewhere, but was never defined AFAICT.)

**Review by dfawley — APPROVED** [review]

File: A21-service-config-error-handling.md (hunk 2 above)
Comment (yashykt) [reply A21-service-config-error-handling.md:47]: Done

File: A21-service-config-error-handling.md (hunk 1 above)
Comment (yashykt) [reply A21-service-config-error-handling.md]: Done

**Review by markdroth — APPROVED** [review]

File: A21-service-config-error-handling.md (hunk 2 above)
Comment (ejona86) [inline A21-service-config-error-handling.md:47]: Shouldn't this be (2) and (2) be renamed to (3)?

**Review by ejona86 — APPROVED** [review]

File: A21-service-config-error-handling.md (hunk 2 above)
Comment (yashykt) [reply A21-service-config-error-handling.md:47]: You mean reordering the two? Will do

Comment (ejona86) [reply A21-service-config-error-handling.md:47]: Yeah, reorder it between the current 1 and 2, since 2 is about validation.

**Review by ejona86 — APPROVED** [review]

**Pull request merged**